pkg/server/v2/server.go

```diff
@@ -50,28 +47,13 @@ type Server interface {
 
 	// Fetch is the universal fetch method.
 	Fetch(context.Context, *discovery.DiscoveryRequest) (*discovery.DiscoveryResponse, error)
-	// StreamHandler is the universal stream method.
-	StreamHandler(stream Stream, typeURL string) error
 }
 
 // Callbacks is a collection of callbacks inserted into the server operation.
 // The callbacks are invoked synchronously.
 type Callbacks interface {
-	// OnStreamOpen is called once an xDS stream is open with a stream ID and the type URL (or "" for ADS).
-	// Returning an error will end processing and close the stream. OnStreamClosed will still be called.
-	OnStreamOpen(context.Context, int64, string) error
-	// OnStreamClosed is called immediately prior to closing an xDS stream with a stream ID.
-	OnStreamClosed(int64)
-	// OnStreamRequest is called once a request is received on a stream.
-	// Returning an error will end processing and close the stream. OnStreamClosed will still be called.
-	OnStreamRequest(int64, *discovery.DiscoveryRequest) error
-	// OnStreamResponse is called immediately prior to sending a response on a stream.
-	OnStreamResponse(int64, *discovery.DiscoveryRequest, *discovery.DiscoveryResponse)
-	// OnFetchRequest is called for each Fetch request. Returning an error will end processing of the
-	// request and respond with an error.
-	OnFetchRequest(context.Context, *discovery.DiscoveryRequest) error
-	// OnFetchResponse is called immediately prior to sending a response.
-	OnFetchResponse(*discovery.DiscoveryRequest, *discovery.DiscoveryResponse)
+	rest.Callbacks
+	sotw.Callbacks
```

Comment:
So i'm not opposed to this, but we need to keep in mind that we are breaking the existing external API so if we merge this, all pre-existing users of go-control-plane will need to update their logic. My incremental PR only adds to functions into the `Callbacks interface{}`: `OnStreamDeltaRequest()` and `OnStreamDeltaResponse()`. I'm good with the change but this is breaking so a release will need to be cut

@kyessenov what do you think?

Reply:
Why, the interface by itself remains the same.

Reply:
Oh oops good point I thought you changed the underlying callbacks too, i'll remove the comment sorry about that!